### PR TITLE
create fs-cloud profile

### PR DESCRIPTION
**Pull request by Aayush-Abhyarthi**
### Description

Replace this text with a summary of the changes in this PR. Include why the changes are needed and context about the changes. List required dependencies. If there is a Git issue for the change, please link to it.

https://github.ibm.com/GoldenEye/issues/issues/5042

### Release required?
Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning).

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

If a release is required, replace this text with information that users need to know about the release. Write the release notes to help users understand the changes, and include information about how to update from the previous version.

Your notes help the merger write the commit message for the PR that is published in the release notes for the module.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

**Comment by Aayush-Abhyarthi** [comment]
Hi @ocofaigh , don't know why this error in CRA is being thrown

**Comment by ocofaigh** [comment]
@Aayush-Abhyarthi This fscloud profile has two required input variables, `existing_kms_instance_guid` and `kms_key_crn`. So that means you need to add these to the cra-config.yaml file like this:
```
# More info about this file at https://github.com/terraform-ibm-modules/common-pipeline-assets/blob/main/.github/workflows/terraform-test-pipeline.md#cra-config-yaml
version: "v1"
CRA_TARGETS:
  - CRA_TARGET: "examples/fscloud"
    CRA_IGNORE_RULES_FILE: "cra-tf-validate-ignore-rules.json"
    PROFILE_ID: "0e6e7b5a-817d-4344-ab6f-e5d7a9c49520" # SCC profile ID (currently set to the FSCloud 1.4.0 profile).
    CRA_ENVIRONMENT_VARIABLES: 
      TF_VAR_existing_kms_instance_guid: "e6dce284-e80f-46e1-a3c1-830f7adff7a9"
      TF_VAR_kms_key_crn:  "crn:v1:bluemix:public:hs-crypto:us-south:a/abac0df06b644a9cabc6e44f55b3880e:e6dce284-e80f-46e1-a3c1-830f7adff7a9:key:76170fae-4e0c-48c3-8ebe-326059ebb533"
```

File: cra-config.yaml [hunk 1]
```diff
@@ -1,11 +1,10 @@
...
-    # SCC_INSTANCE_ID: "" # The SCC instance ID to use to download profile for CRA scan. If not provided, a default global value will be used.
-    # SCC_REGION: "" # The IBM Cloud region that the SCC instance is in. If not provided, a default global value will be used.
-    # PROFILE_ID: "" # The Profile ID input for CRA SCC scan. If not provided, a default global value will be used.
+  - CRA_TARGET: "examples/fscloud"
+    CRA_IGNORE_RULES_FILE: "cra-tf-validate-ignore-rules.json"
+    PROFILE_ID: "0e6e7b5a-817d-4344-ab6f-e5d7a9c49520" # SCC profile ID (currently set to the FSCloud 1.4.0 profile).
+    CRA_ENVIRONMENT_VARIABLES:
+      TF_VAR_existing_kms_instance_guid: "e6dce284-e80f-46e1-a3c1-830f7adff7a9"
+      TF_VAR_kms_key_crn:  "crn:v1:bluemix:public:hs-crypto:us-south:a/abac0df06b644a9cabc6e44f55b3880e:e6dce284-e80f-46e1-a3c1-830f7adff7a9:key:76170fae-4e0c-48c3-8ebe-326059ebb533"
+      TF_VAR_kms_encryption_enabled: true
```
Comment (ocofaigh) [inline cra-config.yaml:10]: remove `TF_VAR_kms_encryption_enabled` - encryption is not optional in fscloud, so this var should not be exposed in fscloud profile

**Review by ocofaigh — CHANGES_REQUESTED** [review]
move the fscloud profile from profiles folder to a modules folder please

File: profiles/fscloud/main.tf [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  existing_kms_instance_guid = var.existing_kms_instance_guid
+  service_credential_names   = var.service_credential_names
+  kms_key_crn                = var.kms_key_crn
+  admin_pass                 = var.admin_pass
+  members                    = var.members
+  users                      = var.users
+  disk_mb                    = var.disk_mb
+  cpu_count                  = var.cpu_count
+  auto_scaling               = var.auto_scaling
+  cbr_rules                  = var.cbr_rules
```
Comment (ocofaigh) [inline profiles/fscloud/main.tf:19]: This is missing `kms_encryption_enabled = true` so hence pipeline is failing

File: cra-config.yaml (hunk 1 above)
Comment (Aayush-Abhyarthi) [reply cra-config.yaml]: removed

File: profiles/fscloud/main.tf (hunk 2 above)
Comment (Aayush-Abhyarthi) [reply profiles/fscloud/main.tf]: added

File: examples/fscloud/version.tf [hunk 3]
```diff
@@ -0,0 +1,10 @@
+terraform {
+  required_version = ">= 1.3.0"
+  required_providers {
+    # Use latest version of provider in non-basic examples to verify latest version works with module
+    ibm = {
+      source  = "IBM-Cloud/ibm"
+      version = ">= 1.54.0"
```
Comment (ocofaigh) [inline examples/fscloud/version.tf:7]: this now need to be updated to >= 1.56.1 since https://github.com/terraform-ibm-modules/terraform-ibm-icd-rabbitmq/pull/22 was merged

File: modules/fscloud/version.tf [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  required_version = ">= 1.3.0"
+  required_providers {
+    # The below tflint-ignore is required because although the below provider is not directly required by this submodule,
+    # it is required by consuming modules, and if not set here, the top level module calling this module will not be
+    # able to set alternative alias for the provider.
+    # tflint-ignore: terraform_unused_required_providers
+    ibm = {
+      source = "IBM-Cloud/ibm"
+      # Use "greater than or equal to" range in modules
+      version = ">= 1.54.0"
```
Comment (ocofaigh) [inline modules/fscloud/version.tf:11]: this now need to be updated to >= 1.56.1 since https://github.com/terraform-ibm-modules/terraform-ibm-icd-rabbitmq/pull/22 was merged

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

File: tests/pr_test.go [hunk 5]
```diff
@@ -47,12 +72,12 @@ func TestRunComplete(t *testing.T) {
...
-		TerraformDir:       completeExampleTerraformDir,
-		Prefix:             "rabbitmq-cmp",
-		ResourceGroup:      resourceGroup,
+		TerraformDir:       "examples/complete",
+		Prefix:             "rabbitmq-upg",
 		BestRegionYAMLPath: regionSelectionPath,
+		ResourceGroup:      resourceGroup,
 		TerraformVars: map[string]interface{}{
-			"rabbitmq_version": "3.11", // Always lock this test into the latest supported RabbitMQ version
+			"rabbitmq_version": "3.11", // Always lock to the lowest supported RabbitMQ version
```
Comment (ocofaigh) [inline tests/pr_test.go:80]: the lowest is 3.9

File: outputs.tf [hunk 6]
```diff
@@ -41,10 +41,10 @@ output "service_credentials_object" {
...
+  value       = length(var.service_credential_names) > 0 ? nonsensitive(ibm_resource_key.service_credentials[keys(var.service_credential_names)[0]].credentials["connection.https.hosts.0.hostname"]) : length(var.users) > 0 ? nonsensitive(flatten(data.ibm_database_connection.database_connection[0].https[0].hosts[0].hostname)) : null
 }
 
 output "port" {
   description = "Database port. Only contains value when var.service_credential_names or var.users are set."
-  value       = length(var.service_credential_names) > 0 ? nonsensitive(ibm_resource_key.service_credentials[keys(var.service_credential_names)[0]].credentials["connection.https.hosts.0.port"]) : length(var.users) > 0 ? data.ibm_database_connection.database_connection[0].https[0].hosts[0].port : null
+  value       = length(var.service_credential_names) > 0 ? nonsensitive(ibm_resource_key.service_credentials[keys(var.service_credential_names)[0]].credentials["connection.https.hosts.0.port"]) : length(var.users) > 0 ? nonsensitive(flatten(data.ibm_database_connection.database_connection[0].https[0].hosts[0].port)) : null
```
Comment (ocofaigh) [inline outputs.tf:49]: can you please revert this change - the code in main branch is correct for these outputs

File: examples/fscloud/variables.tf [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+variable "existing_kms_instance_guid" {
+  description = "The GUID of the Hyper Protect Crypto service in which the key specified in var.kms_key_crn is coming from"
+  type        = string
+}
+
+variable "kms_key_crn" {
+  type        = string
+  description = "The root key CRN of a Hyper Protect Crypto Service (HPCS) that you want to use for disk encryption. See https://cloud.ibm.com/docs/cloud-databases?topic=cloud-databases-hpcs&interface=ui for more information on integrating HPCS with MongoDB instance."
```
Comment (ocofaigh) [inline examples/fscloud/variables.tf:38]: MongoDB -> RabbitMQ

File: examples/fscloud/main.tf [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+  total_ipv4_address_count = 256
+  resource_group           = module.resource_group.resource_group_id
+}
+
+##############################################################################
+# Create CBR Zone
+##############################################################################
+module "cbr_zone" {
+  source           = "terraform-ibm-modules/cbr/ibm//cbr-zone-module"
+  version          = "1.6.1"
```
Comment (ocofaigh) [inline examples/fscloud/main.tf:42]: Can you use version 1.8.0 - the `cbr-zone-module` is now moved into a modules folder, so you will need to also update the source url

File: examples/fscloud/README.md [hunk 9]
```diff
@@ -0,0 +1,19 @@
+# Financial Services Cloud profile example with autoscaling enabled
+
+An end-to-end example that uses the [Profile for IBM Cloud Framework for Financial Services](../../profiles/fscloud/) to deploy an instance of IBM Cloud Databases for RabbitMQ.
```
Comment (ocofaigh) [inline examples/fscloud/README.md:3]: the hyperlink is wrong `profiles/fscloud` -> `modules/fscloud`

File: outputs.tf [hunk 10]
```diff
@@ -41,7 +41,7 @@ output "service_credentials_object" {
 
 output "hostname" {
   description = "Database hostname. Only contains value when var.service_credential_names or var.users are set."
-  value       = length(var.service_credential_names) > 0 ? nonsensitive(ibm_resource_key.service_credentials[keys(var.service_credential_names)[0]].credentials["connection.https.hosts.0.hostname"]) : length(var.users) > 0 ? data.ibm_database_connection.database_connection[0].https[0].hosts[0].hostname : null
+  value       = length(var.service_credential_names) > 0 ? nonsensitive(ibm_resource_key.service_credentials[keys(var.service_credential_names)[0]].credentials["connection.https.hosts.0.hostname"]) : length(var.users) > 0 ? nonsensitive(flatten(data.ibm_database_connection.database_connection[0].https[0].hosts[0].hostname)) : null
```
Comment (ocofaigh) [inline outputs.tf:44]: Please revert this change - you are re-introducing a bug here. The code in main branch is fine

File: tests/pr_test.go [hunk 11]
```diff
@@ -63,30 +88,6 @@ func TestRunComplete(t *testing.T) {
...
-	options.SkipTestTearDown = true
-	output, err := options.RunTestConsistency()
-	assert.Nil(t, err, "This should not have errored")
-	assert.NotNil(t, output, "Expected some output")
-	outputs := terraform.OutputAll(options.Testing, options.TerraformOptions)
-	expectedOutputs := []string{"port", "hostname"}
-	_, outputErr := testhelper.ValidateTerraformOutputs(outputs, expectedOutputs...)
-	assert.NoErrorf(t, outputErr, "Some outputs not found or nil")
-	options.TestTearDown()
-}
```
Comment (ocofaigh) [inline tests/pr_test.go:75]: Can you please add back the extra test logic here to check that port and hostname have values

File: modules/fscloud/README.md [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+
+:exclamation: **Exception:** The Databases for RabbitMQ service is not yet Financial services validated. Therefore, the infrastructure that is deployed by this module is also not validated with the Framework for Financial Services. For more information, see the list of [Financial Services Validated services](https://cloud.ibm.com/docs/framework-financial-services?topic=framework-financial-services-vpc-architecture-about#financial-services-validated-services).
+
+The default values in this profile were scanned by [IBM Code Risk Analyzer (CRA)](https://cloud.ibm.com/docs/code-risk-analyzer-cli-plugin?topic=code-risk-analyzer-cli-plugin-cra-cli-plugin#terraform-command) for compliance with the IBM Cloud Framework for Financial Services profile that is specified by the IBM Security and Compliance Center. The scan passed for all applicable goals with one exception:
+
+> rule-beb7b289-706b-4dc0-b01d-b1d15d4331e3: Check whether RabbitMQ network access is restricted to a specific IP range.
```
Comment (ocofaigh) [inline modules/fscloud/README.md:9]: Is this correct? I don't see this rule being ignored in `cra-tf-validate-ignore-rules.json` and I don't see it failing in the pipeline either?

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see latest comments please @Aayush-Abhyarthi

File: outputs.tf [hunk 13]
```diff
@@ -41,7 +41,7 @@ output "service_credentials_object" {
 
 output "hostname" {
   description = "Database hostname. Only contains value when var.service_credential_names or var.users are set."
-  value       = length(var.service_credential_names) > 0 ? nonsensitive(ibm_resource_key.service_credentials[keys(var.service_credential_names)[0]].credentials["connection.https.hosts.0.hostname"]) : length(var.users) > 0 ? nonsensitive(flatten(data.ibm_database_connection.database_connection[0].https[0].hosts[0].hostname)) : null
+  value       = length(var.service_credential_names) > 0 ? nonsensitive(ibm_resource_key.service_credentials[keys(var.service_credential_names)[0]].credentials["connection.https.hosts.0.hostname"]) : length(var.users) > 0 ? data.ibm_database_connection.database_connection[0].https[0].hosts[0].hostname : null
```
Comment (ocofaigh) [inline outputs.tf:44]: why are you making changes to this? Just leave it as it is in main branch

File: cra-tf-validate-ignore-rules.json [hunk 14]
```diff
@@ -5,6 +5,12 @@
         "description": "Check whether Flow Logs for VPC are enabled",
         "ignore_reason": "This rule is not relevant to the module itself, just the VPC resource is used in the example that is scanned",
         "is_valid": false
+      },
+      {
+        "scc_rule_id": "rule-beb7b289-706b-4dc0-b01d-b1d15d4331e3",
+        "description": "Check whether RabbitMQ network access is restricted to a specific IP range",
+        "ignore_reason": "This module supports restricting network access using Context Based Restrictions (CBRs), however SCC does not yet support scanning for CBR rules, hence the rule currently fails. SCC CBR support is being tracked in AHA SCC-961",
+        "is_valid": true
```
Comment (ocofaigh) [inline cra-tf-validate-ignore-rules.json:13]: why are you adding this? I don't see this rule failing in the pipeline??

**Review by ocofaigh — CHANGES_REQUESTED** [review]
The fscloud test failed with this:
```
Error Trace:	/__w/terraform-ibm-icd-rabbitmq/terraform-ibm-icd-rabbitmq/tests/pr_test.go:69
        	Error:      	Received unexpected error:
        	            	Output: 'port' was not found
        	            	Output: 'hostname' was not found
        	Test:       	TestRunFSCloudExample
        	Messages:   	Some outputs not found or nil
```

This is because you are not passing an admin username or creating service credentials in the fscloud example. Either do this, or move the test assertion to the complete upgrade test

**Comment by ocofaigh** [comment]
@Aayush-Abhyarthi can you please rebase with master and resolve conflicts?

File: modules/fscloud/README.md (hunk 12 above)
Comment (ocofaigh) [reply modules/fscloud/README.md:9]: @Aayush-Abhyarthi I don't see this rule failing - can you confirm?

**Review by ocofaigh — COMMENTED** [review]
@Aayush-Abhyarthi Just one [comment](https://github.com/terraform-ibm-modules/terraform-ibm-icd-rabbitmq/pull/18/files#r1314927105) left

File: tests/pr_test.go [hunk 15]
```diff
@@ -37,60 +37,70 @@ func TestMain(m *testing.M) {
...
+					"name":     "testuser",
+					"password": randomPass, // pragma: allowlist secret
+					"type":     "database",
+				},
+			},
+			"admin_pass": randomPass,
 		},
 	})
-
+	options.SkipTestTearDown = true
```
Comment (daniel-butler-irl) [inline tests/pr_test.go:95]: If you just need the output you no longer need to skip the tearDown. You can access the outputs with `options.LastTestTerraformOutputs`

Comment (ocofaigh) [reply tests/pr_test.go:95]: yes good point Daniel, now that we have bumped to latest test wrapper version we can leverage this

Comment (Aayush-Abhyarthi) [reply tests/pr_test.go:95]: Hi @daniel-butler-irl , `options.LastTestTerraformOutputs` is not available in this package. Which package should I use?

**Review by ocofaigh — CHANGES_REQUESTED** [review]
@Aayush-Abhyarthi Can you please remove all variable validation from `modules/fscloud/variables.tf` - since the fscloud submodule calls the root level module, the validation in the root level variables.tf will be used, so no need to duplicate it and maintain in two places

File: modules/fscloud/variables.tf [hunk 16]
```diff
@@ -0,0 +1,159 @@
...
+  description = "(Optional, list) List of CBR rules to create"
+  default     = []
+  # Validation happens in the rule module
+}
+
+variable "access_tags" {
+  type        = list(string)
+  description = "A list of access tags to apply to the rabbitmq instance created by the module, see https://cloud.ibm.com/docs/account?topic=account-access-tags-tutorial for more details"
+  default     = []
+}
```
Comment (ocofaigh) [inline modules/fscloud/variables.tf:159]: Please remove all the variable validation from the fscloud submodule. The validation in the root level module will catch any issue, so no need to duplicate it

File: examples/fscloud/version.tf [hunk 17]
```diff
@@ -0,0 +1,10 @@
+terraform {
+  required_version = ">= 1.3.0"
```
Comment (ocofaigh) [inline examples/fscloud/version.tf:2]: please update this to `>= 1.3.0, <1.6.0`

File: examples/fscloud/version.tf [hunk 18]
```diff
@@ -0,0 +1,10 @@
...
+  required_providers {
+    # Use latest version of provider in non-basic examples to verify latest version works with module
+    ibm = {
+      source  = "IBM-Cloud/ibm"
+      version = ">= 1.56.1"
```
Comment (ocofaigh) [inline examples/fscloud/version.tf:7]: update this to `>= 1.56.1, <2.0.0`

File: modules/fscloud/version.tf [hunk 19]
```diff
@@ -0,0 +1,14 @@
+terraform {
+  required_version = ">= 1.3.0"
```
Comment (ocofaigh) [inline modules/fscloud/version.tf:2]: `>= 1.3.0, <1.6.0`

File: modules/fscloud/version.tf [hunk 20]
```diff
@@ -0,0 +1,14 @@
...
+  required_providers {
+    # The below tflint-ignore is required because although the below provider is not directly required by this submodule,
+    # it is required by consuming modules, and if not set here, the top level module calling this module will not be
+    # able to set alternative alias for the provider.
+    # tflint-ignore: terraform_unused_required_providers
+    ibm = {
+      source = "IBM-Cloud/ibm"
+      # Use "greater than or equal to" range in modules
+      version = ">= 1.56.1"
```
Comment (ocofaigh) [inline modules/fscloud/version.tf:11]: `>= 1.56.1, <2.0.0`

File: variables.tf [hunk 21]
```diff
@@ -60,9 +60,9 @@ variable "rabbitmq_version" {
   validation {
     condition = anytrue([
       var.rabbitmq_version == null,
-      var.rabbitmq_version == "3.11"
+      var.rabbitmq_version == "3.9"
     ])
-    error_message = "Version must be 3.11. If no value passed, the current ICD preferred version is used."
+    error_message = "Version must be 3.9. If no value passed, the current ICD preferred version is used."
```
Comment (ocofaigh) [inline variables.tf:65]: why are you changing this to 3.9? 3.9 is no longer supported. Please revert this change. Only 3.11 is currently a supported version

File: tests/pr_test.go [hunk 22]
```diff
@@ -48,12 +77,12 @@ func TestRunComplete(t *testing.T) {
...
-		TerraformDir:       completeExampleTerraformDir,
-		Prefix:             "rabbitmq-cmp",
-		ResourceGroup:      resourceGroup,
+		TerraformDir:       "examples/complete",
+		Prefix:             "rabbitmq-upg",
 		BestRegionYAMLPath: regionSelectionPath,
+		ResourceGroup:      resourceGroup,
 		TerraformVars: map[string]interface{}{
-			"rabbitmq_version": "3.11", // Always lock this test into the latest supported RabbitMQ version
+			"rabbitmq_version": "3.9", // Always lock to the lowest supported RabbitMQ version
```
Comment (ocofaigh) [inline tests/pr_test.go:85]: change this back - only 3.11 is currently supported

File: modules/fscloud/README.md (hunk 12 above)
Comment (ocofaigh) [reply modules/fscloud/README.md:9]: @Aayush-Abhyarthi There is no failing rule in the CRA scan - why is still documented as a failing rule?

**Review by ocofaigh — CHANGES_REQUESTED** [review]
@Aayush-Abhyarthi Can you please address https://github.com/terraform-ibm-modules/terraform-ibm-icd-rabbitmq/pull/18/files#r1314927105

File: tests/pr_test.go [hunk 23]
```diff
@@ -37,55 +37,60 @@ func TestMain(m *testing.M) {
...
 	})
+
 	options.SkipTestTearDown = true
 	output, err := options.RunTestConsistency()
 	assert.Nil(t, err, "This should not have errored")
 	assert.NotNil(t, output, "Expected some output")
-	outputs := terraform.OutputAll(options.Testing, options.TerraformOptions)
-	expectedOutputs := []string{"port", "hostname"}
-	_, outputErr := testhelper.ValidateTerraformOutputs(outputs, expectedOutputs...)
-	assert.NoErrorf(t, outputErr, "Some outputs not found or nil")
```
Comment (ocofaigh) [inline tests/pr_test.go:74]: since you have moved this assertion to the `TestRunCompleteUpgradeExample` test, you no longer need to call `SkipTestTearDown` and `TestTearDown`

File: examples/fscloud/main.tf [hunk 24]
```diff
@@ -0,0 +1,87 @@
+##############################################################################
+# Resource Group
+##############################################################################
+
+module "resource_group" {
+  source  = "terraform-ibm-modules/resource-group/ibm"
+  version = "1.0.6"
```
Comment (ocofaigh) [inline examples/fscloud/main.tf:7]: bump to 1.1.0

File: examples/fscloud/main.tf [hunk 25]
```diff
@@ -0,0 +1,87 @@
...
+  total_ipv4_address_count = 256
+  resource_group           = module.resource_group.resource_group_id
+}
+
+##############################################################################
+# Create CBR Zone
+##############################################################################
+module "cbr_zone" {
+  source           = "terraform-ibm-modules/cbr/ibm//modules/cbr-zone-module"
+  version          = "1.8.0"
```
Comment (ocofaigh) [inline examples/fscloud/main.tf:42]: bump to 1.15.1

File: modules/fscloud/variables.tf [hunk 26]
```diff
@@ -0,0 +1,147 @@
...
+
+variable "existing_kms_instance_guid" {
+  description = "The GUID of the Hyper Protect Crypto Services instance."
+  type        = string
+}
+
+variable "admin_pass" {
+  type        = string
+  description = "The password for the database administrator. If the admin password is null then the admin user ID cannot be accessed. More users can be specified in a user block. The admin password must be in the range of 10-32 characters."
```
Comment (ocofaigh) [inline modules/fscloud/variables.tf:58]: Can you update the last sentence to:

The admin password must be in the range of 10-32 characters and cannot start with a special character.

**Review by ocofaigh — CHANGES_REQUESTED** [review]
We are nearly there @Aayush-Abhyarthi - please address final few comments. Hoping to get this merged today. Thanks

**Comment by ocofaigh** [comment]
@daniel-butler-irl @toddgiguere Any idea why test is failing with this:
```
TestRunCompleteUpgradeExample 2023-10-25T06:23:24Z tests.go:399: PR Branch: HEAD
    tests.go:403: Detected the string "BREAKING CHANGE" or "SKIP UPGRADE TEST" used in commit message, skipping upgrade Test.
--- FAIL: TestRunCompleteUpgradeExample (4.24s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xdfbb02]

goroutine 7 [running]:
testing.tRunner.func1.2({0x172b820, 0x2d7e050})
	/usr/local/go/src/testing/testing.go:[154](https://github.com/terraform-ibm-modules/terraform-ibm-icd-rabbitmq/actions/runs/6631100502/job/18029157416#step:7:155)5 +0x238
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1548 +0x397
panic({0x172b820?, 0x2d7e050?})
	/usr/local/go/src/runtime/panic.go:914 +0x21f
github.com/gruntwork-io/terratest/modules/terraform.GetCommonOptions(0x1a4572a?, {0xc000b923c0?, 0x0?, 0x0?})
	/go/pkg/mod/github.com/gruntwork-io/terratest@v0.46.1/modules/terraform/cmd.go:46 +0x22
github.com/gruntwork-io/terratest/modules/terraform.RunTerraformCommandAndGetStdoutE({0x1efbd18, 0xc0001b5a00}, 0xa?, {0xc000b923c0?, 0xc000b93020?, 0xc000002c00?})
	/go/pkg/mod/github.com/gruntwork-io/terratest@v0.46.1/modules/terraform/cmd.go:92 +0x46
github.com/gruntwork-io/terratest/modules/terraform.OutputJsonE({0x1efbd18, 0xc0001b5a00}, 0x0?, {0x0, 0x0})
	/go/pkg/mod/github.com/gruntwork-io/terratest@v0.46.1/modules/terraform/output.go:282 +0xfb
github.com/gruntwork-io/terratest/modules/terraform.OutputForKeysE({0x1efbd18?, 0xc0001b5a00?}, 0x0?, {0x0, 0x0, 0x0?})
	/go/pkg/mod/github.com/gruntwork-io/terratest@v0.46.1/modules/terraform/output.go:310 +0x46
github.com/gruntwork-io/terratest/modules/terraform.OutputAllE(...)
	/go/pkg/mod/github.com/gruntwork-io/terratest@v0.46.1/modules/terraform/output.go:349
github.com/gruntwork-io/terratest/modules/terraform.OutputAll({0x1efbd18?, 0xc0001b5a00}, 0x1a371c5?)
	/go/pkg/mod/github.com/gruntwork-io/terratest@v0.46.1/modules/terraform/output.go:342 +0x2a
command-line-arguments.TestRunCompleteUpgradeExample(0xc0001b5a00)
	/__w/terraform-ibm-icd-rabbitmq/terraform-ibm-icd-rabbitmq/tests/pr_test.go:100 +0x479
testing.tRunner(0xc0001b5a00, 0x1afe568)
	/usr/local/go/src/testing/testing.go:[159](https://github.com/terraform-ibm-modules/terraform-ibm-icd-rabbitmq/actions/runs/6631100502/job/18029157416#step:7:160)5 +0xff
created by testing.(*T).Run in goroutine 1
	/usr/local/go/src/testing/testing.go:[164](https://github.com/terraform-ibm-modules/terraform-ibm-icd-rabbitmq/actions/runs/6631100502/job/18029157416#step:7:165)8 +0x3ad
FAIL	command-line-arguments	4.340s
FAIL
make: *** [Makefile:126: run-tests] Error 1
```

**Comment by daniel-butler-irl** [comment]
the error is with the test, it is skipped but you still try to access the outputs. All that code should be in the if !options.UpgradeTestSkipped block

**Review by ocofaigh — APPROVED** [review]
